### DISCUSSION TITLE
Push notifications

### DISCUSSION THREAD

**Discussion by mateus-holanda** [diego3g/rsxp-2023#11] [Ideas]
Caso dê tempo para essa primeira versão, além das notificações dentro do próprio app poderia ter também notificações do tipo push na tela do usuário, alertando sobre speakers confirmados, lembrete quando a data estiver se aproximando ou quando um amigo aceitar um convite e comprar um ingresso através do link pessoal do usuário. Isso poderia incentivar o usuário a estar continuamente verificando o app e checando atualizações ao invés de instalar uma vez e esquecer dele até o dia do evento.

**Comment by melquisedeque-magalhaes** [comment]
Acho uma boa ideia, poderíamos usar o onesignal para dash de envio de notificações, e bem tranquilo de integrar ele com ios e Android no react native, já usei ele e é muito bom.

**Reply by pedroarm** [reply]
É uma boa ideia. Também é possível usar o Firebase Cloud Messaging.

**Reply by Ramiro-Rfn** [reply]
Também pensei em oneSignal logo quando se falou de notificações push

**Comment by orodrigogo** [comment]
Boa @mateus-holanda e @melquisedeque-magalhaes, muito massa as contribuições.

Eu até vejo a possibilidade de separar as notificações em local e push. Por exemplo:

- **Notificação Local:** lembrete das palestras que eu escolhi participar e não quero perder.
- **Push Notifications:** notificações em geral.

Que tal? 🤗

**Reply by melquisedeque-magalhaes** [reply]
Boa ideia @rodrigorgtic curti essa parada de notificações local, muito massa 💯

**Comment by carlosdavid0** [comment]
Pensando em notificações locais, notifee é uma ótima opção. Porém, quando falamos em push, qual a melhor alternativa?  OneSignal? Firebase? Existe outros? Se sim, quais? 


**Edit:** Podemos usar o SNS da AWS.

**Reply by orodrigogo** [reply]
Todas são excelentes opções, eu gosto bastante do Notifee e do Onesignal. Mas, para a v1 podemos utilizar o Expo Notification que permite trabalhar com Push e Local Notifications. Tem uma interface bem legal para trabalhar. Que tal?

**Reply by carlosdavid0** [reply]
Interessante, super concordo.

**Reply by leoferreiralima** [reply]
@rodrigorgtic concordo em ir com o expo notification, ótima ideia

**Reply by JonathanSeehagen** [reply]
@rodrigorgtic, com o Expo Notification, será necessário criar no backend os serviços para o armazenamento dos tokens e para disparar as Push Notifications, correto?
